web/assets/javascripts/metrics.js

```diff
@@ -60,6 +61,17 @@ class JobMetricsOverviewChart extends BaseChart {
     };
   }
 
+  update(options) {
```

Comment:
@adamlogic I tried to write an update method based on the documentation for Chart.js but kind of got stuck.
How would you implement it? Would like to be able to update the datasets of existing chart.

Reply:
You should be able to mutate `this.chart.data.datasets` then call `this.chart.update()`.

Take a look at `RealtimeChart` for how I'm live-updating the chart on the dashboard. You might be able to extract a lot of that into something that can be used by both `dashboard-charts.js` and `metrics.js`.

https://github.com/adamlogic/sidekiq/blob/0039b849c061ab43f29019d2dd2aeb0310705799/web/assets/javascripts/dashboard-charts.js#L76-L82

Reply:
Thanks, will check it out! 🙂